From 6ec1590129b1855ce3366d53636f6a6272bbdb86 Mon Sep 17 00:00:00 2001
From: Bruno Thomsen <bth@kamstrup.dk>
Date: Thu, 25 Sep 2014 15:49:40 +0200
Subject: [PATCH] bash-3.2: patch 50

Bash-Release: 3.2
Patch-ID: bash32-050

Bug-Reported-by:	Jan Hnatek <Jan.Hnatek@Sun.COM>
Bug-Reference-ID:	<4A44991F.8010005@sun.com>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2009-06/msg00084.html

Bug-Description:

On systems where mbrtowc() returns -2 when passed a length argument with
value 0, when using a multibyte locale, Readline's emacs-mode forward-char
at the end of a line will leave the point beyond the end of the line.

Signed-off-by: Bruno Thomsen <bth@kamstrup.dk>
---
 lib/readline/mbutil.c | 4 +---
 patchlevel.h          | 2 +-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/readline/mbutil.c b/lib/readline/mbutil.c
index 0f8191c..6d3da95 100644
--- a/lib/readline/mbutil.c
+++ b/lib/readline/mbutil.c
@@ -131,12 +131,10 @@ _rl_find_next_mbchar_internal (string, seed, count, find_non_zero)
   if (find_non_zero)
     {
       tmp = mbrtowc (&wc, string + point, strlen (string + point), &ps);
-      while (tmp > 0 && wcwidth (wc) == 0)
+      while (MB_NULLWCH (tmp) == 0 && MB_INVALIDCH (tmp) == 0 && wcwidth (wc) == 0)
 	{
 	  point += tmp;
 	  tmp = mbrtowc (&wc, string + point, strlen (string + point), &ps);
-	  if (MB_NULLWCH (tmp) || MB_INVALIDCH (tmp))
-	    break;
 	}
     }
 
diff --git a/patchlevel.h b/patchlevel.h
index 19e5eca..37480dc 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 49
+#define PATCHLEVEL 50
 
 #endif /* _PATCHLEVEL_H_ */
-- 
1.9.1

